webrender_api/src/font.rs

```diff
@@ -153,31 +153,61 @@ impl SubpixelPoint {
     }
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Hash, Eq, PartialEq, PartialOrd, Ord, Serialize)]
+pub struct GlyphOptions {
+    // These are currently only used on windows for dwrite fonts.
+    pub use_embedded_bitmap: bool,
+    pub force_gdi_rendering: bool,
+}
+
 #[derive(Clone, Hash, PartialEq, Eq, Debug, Deserialize, Serialize, Ord, PartialOrd)]
-pub struct GlyphKey {
+pub struct FontInstanceKey {
     pub font_key: FontKey,
     // The font size is in *device* pixels, not logical pixels.
     // It is stored as an Au since we need sub-pixel sizes, but
     // can't store as a f32 due to use of this type as a hash key.
     // TODO(gw): Perhaps consider having LogicalAu and DeviceAu
     //           or something similar to that.
     pub size: Au,
-    pub index: u32,
     pub color: ColorU,
-    pub subpixel_point: SubpixelPoint,
+    pub render_mode: FontRenderMode,
+    pub glyph_options: Option<GlyphOptions>,
 }
 
-impl GlyphKey {
+impl FontInstanceKey {
     pub fn new(font_key: FontKey,
                size: Au,
-               color: ColorF,
-               index: u32,
+               mut color: ColorF,
+               render_mode: FontRenderMode,
+               glyph_options: Option<GlyphOptions>) -> FontInstanceKey {
+        // In alpha/mono mode, the color of the font is irrelevant.
```

Comment:
how is the color relevant to the font key when in `Subpixel` mode?
My first guess would be that the color values produced by the sub-pixel AA would just be multiplied by the given text color, which means that the rasterization is done independent of text color.